File: python/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        stored = {}
        for idx, num in enumerate(nums):
            diff = target - num
            if diff in stored:
                return [stored[diff], idx]
            stored[num] = idx

File: go/two_sum/two_sum.go
package kata

// ::KATA START::
func twoSum(nums []int, target int) []int {
	result := make([]int, 2)
	hashSet := map[int]int{}

	for idx, num := range nums {
		diff := target - num
		if val, ok := hashSet[num]; ok {
			result[0] = val
			result[1] = idx
			break
		}
		hashSet[diff] = idx
	}

	return result
}

// ::KATA END::
